python/cudf/cudf/core/column/interval.py

```diff
@@ -0,0 +1,58 @@
+import pyarrow as pa
+import cudf
+from cudf.core.column import StructColumn
+
+
+class IntervalColumn(StructColumn):
+    def __init__(
+        self,
+        dtype,
+        mask=None,
+        size=None,
+        offset=0,
+        null_count=None,
+        children=(),
+        closed="right",
+    ):
+
+        super().__init__(
+            data=None,
+            dtype=dtype,
+            mask=mask,
+            size=size,
+            offset=offset,
+            null_count=null_count,
+            children=children,
+        )
+        self.closed = closed
```

Comment:
Do we want someone to be able to changed the `closed` value of a Column? If not we should set it as `self._closed = closed` and then create a property accessor for `self.closed`.

We should also validate that a valid value is set for `closed`.

Reply:
This makes sense :D I'll make changes